### PR TITLE
[jit] Clean up casting ops

### PR DESCRIPTION
**Pull request by zdevito**
This removes FloatToInt style names replacing it with just the destination
name (e.g. FloatToInt -> Float). This makes it more consistent with the
syntax and makes it easier to add type conversions (just add a new
prim::Int op, for instance).

None of these ops get serialized so this should not effect loading of
old models.

### CODE REVIEW

File: torch/csrc/jit/passes/shape_analysis.cpp [hunk 1]
```diff
@@ -386,13 +386,15 @@ class ShapePropagator {
...
       case prim::ImplicitTensorToNum:
-      case prim::TensorToNum:
+      case prim::Bool:
+      case prim::Int:
+      case prim::Float:
         return; // correct num type is already set
       case prim::NumToTensor: {
-        if (node->input()->type()->isSubtypeOf(IntType::get())) {
+        TypePtr typ = node->input()->type();
+        if (typ->isSubtypeOf(IntType::get()) | typ->isSubtypeOf(BoolType::get())) {
```
Comment (eellison) [inline torch/csrc/jit/passes/shape_analysis.cpp]: Is this intended to be `|` and not `||` ?

File: torch/csrc/jit/register_prim_ops.cpp [hunk 2]
```diff
@@ -78,25 +78,26 @@ RegisterOperators reg({
           };
         }),
     Operator(
-        "prim::TensorToNum(Tensor a) -> Scalar",
+        "prim::Int(Tensor a) -> int",
+        [](const Node* node) -> Operation {
+          return [](Stack& stack) {
+            at::Tensor a;
+            pop(stack, a);
+            at::OptionalDeviceGuard guard(device_of(a));
```
Comment (eellison) [inline torch/csrc/jit/register_prim_ops.cpp]: Why is this device guard needed ? There isn't a new tensor being created

File: torch/csrc/jit/passes/shape_analysis.cpp [hunk 3]
```diff
@@ -386,13 +386,15 @@ class ShapePropagator {
...
           node->output()->setType(TensorType::create(at::kLong, at::kCPU, 0));
```
Comment (eellison) [inline torch/csrc/jit/passes/shape_analysis.cpp]: I don't know when bool will get converted to Tensor, but i am surprised the tensor type is not kByte. If you create a tensor with: torch.tensor(True) the resulting type is ByteTensor.

Comment (zdevito) [reply torch/csrc/jit/passes/shape_analysis.cpp]: I am copying what was done in BoolToTensor below, which does turn it into a kLong. This behavior is likely not desired, but fixing it is a separate issue from this patch, which is suppose to be a refactor and not a behavior change.

File: torch/csrc/jit/register_prim_ops.cpp (hunk 2 above)
Comment (zdevito) [reply torch/csrc/jit/register_prim_ops.cpp]: Looks like old code...

**Review by eellison — APPROVED** [review]
LGTM. Maybe someone with better knowledge of tracer maybe good take a look as well.

**Pull request closed without merging**